File: 1.two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ans;
        unordered_map<int, int> s;
        for(int i = 0; i < nums.size(); i++) {
            if(s.find(nums[i]) == s.end()) {
                s[target - nums[i]] = i;
            }
            else {
                ans.push_back(s[nums[i]]);
                ans.push_back(i);
                break;
            }
        }
        return ans;
    }
};

File: fb/1.two-sum.java
class Solution {
  public int[] twoSum(int[] nums, int target) {
    Map<Integer, Integer> idxMap = new HashMap<>();
    for (int i = 0; i< nums.length; i++) {
      int complement = target - nums[i];
      if (idxMap.containsKey(complement)) {
        return new int[]{i, idxMap.get(complement)};
      }
      idxMap.put(nums[i], i);
    }
    return null;
  }
}
